src/common/scheduler/scheduler.go

```diff
@@ -254,7 +236,17 @@ func (sch *Scheduler) UnSchedule(policyName string) error {
 	}
 
 	//Unschedule the policy.
-	sch.unscheduleQueue <- policyName
+	//Find the watcher.
+	watcher := sch.policies.Remove(policyName)
+	if watcher != nil && watcher.IsRunning() {
+		watcher.Stop()
+
+		//Update stats and log info.
+		log.Infof("Policy %s is unscheduled", policyName)
+		sch.statChan <- &StatItem{statUnSchedulePolicy, 1, nil}
+	} else {
+		return errors.New("Inconsistent policy status")
```

Comment:
probably log a warning saying the watcher of policy:%s is not running, something maybe wrong, and return a nil?
because the same policy can be rescheduled.